test/e2e/far_e2e_test.go

```diff
@@ -49,25 +52,34 @@ var _ = Describe("FAR E2e", func() {
 		fmt.Printf("\ncluster name: %s and PlatformType: %s \n", string(clusterPlatform.Name), string(clusterPlatform.Status.PlatformStatus.Type))
 	})
 
-	Context("fence agent", func() {
+	Context("fence agent - fence_aws or fence_ipmilan", func() {
+		var (
+			nodeBootTimeBefore   time.Time
+			errBoot              error
+			testNodeName         string
+			nodeIdentifierPrefix string
+			testNodeID           string
+		)
 		BeforeEach(func() {
 			nodes := &corev1.NodeList{}
 			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{})).ToNot(HaveOccurred())
 			if len(nodes.Items) <= 1 {
 				Fail("there is one or less available nodes in the cluster")
 			}
-			//TODO: Randomize the node selection
-			// run FA on the first node - a master node
+			//TODO: Randomize the node selection & verify valid index
+			// run FA on the fourth node - a worker node
 			nodeObj := nodes.Items[nodeIndex]
```

Comment:
Assuming the 4th node would be a worker node (or even that there will be at least 4 nodes) seems a bit risky.
I suggest either  modify the initial fetching of nodes to retrieve worker nodes  (on `k8sClient.List(...  &client.ListOptions{}..` or just filter out the results

Reply:
It is small and important change :D